### DISCUSSION TITLE
Using typer in verb-noun vs noun-verb (with 3+... layers? to it)

### DISCUSSION THREAD

**Discussion by gwinans** [fastapi/typer#697] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import typer

app = typer.Typer()
get_app = typer.Typer()
app.add_typer(get_app, name="get")

@get_app.command()
def admin():
    typer.echo("Admin ...")

if __name__ == "__main__":
    app()
```

### Description

**Quick preamble:**
I'm painfully new to python and more of a surface-scratcher when it comes to programming in general. I'm an old-school sysadmin / DBA that's trying to round-out his skillset.

If I'm asking what amounts to a stupid question, mea culpa.

That said, I couldn't find any examples in the documentation or in the Discussions section on this topic. I recognize I may simply not *know* what to search for.

**The issue:**

My broad goal is to create a tool using the verb-noun command format as such:

```
proxysqlctl get admin [users|variables] --flags ...
proxysqlctl create mysql [server|user] --flags ...
```

I've had no issue getting to:

`proxysqlctl get admin` 

... but I cannot figure out / understand / conceptualize how to add that 3rd+ layer.

Thank you all so much for any help. Typer has been amazing with some of my other one-off tools.

### Operating System

Linux

### Operating System Details

This is not relevant.

### Typer Version

0.7.0

### Python Version

3.12.0

### Additional Context

Best I can offer in terms of context is this roughed-out draft design:

https://github.com/gwinans/proxysqlctl/wiki/Commandline-Design-Idea(s)

Ignore the repo, it's mostly me fooling around.

**Comment by pmav99** [comment, marked as answer]
The tutorial has information for nested subcommands: https://typer.tiangolo.com/tutorial/subcommands/nested-subcommands/
Unless there is some bug, it shouldn't be too difficult to extend it to more levels.

That being said, from a code point of view it might be simpler to just go with:
```
proxysqlctl get admin-users
proxysqlctl get admin-variables
proxysqlctl get mysql-servers
proxysqlctl get mysql-users
proxysqlctl get mysql-variables
...
```
Furthermore, fewer nested levels would probably make it easier for users to discover the available subcommands, too. 

With additional levels you need to do issue extra commands:
```
proxysql get --help              # Ah! there is a mysql sub-subcommand
proxysql get mysql --help        # Ah! there is a users sub-sub-subcommand
proxysql get mysql users --help  # Ah! there is this option that I can use
```

**Reply by gwinans** [reply]
Fair take - simpler set of commands.

I tried crusing through the nested subcommands tutorial, but I just could not make heads or tails of it. I'll re-eval.
